modules/sqs/src/main/scala/vinyldns/sqs/queue/SqsMessage.scala

```diff
@@ -0,0 +1,38 @@
+package vinyldns.sqs.queue
+import java.util.Base64
+
+import cats.implicits._
+import com.amazonaws.services.sqs.model.Message
+import vinyldns.core.domain.ZoneCommand
+import vinyldns.core.protobuf.ProtobufConversions
+import vinyldns.core.queue.CommandMessage
+import vinyldns.proto.VinylDNSProto
+import vinyldns.sqs.queue.SqsMessageQueue.EmptySqsMessageContents
+import vinyldns.sqs.queue.SqsMessageType.{SqsRecordSetChangeMessage, SqsZoneChangeMessage}
+
+final case class SqsMessage(receiptHandle: String, command: ZoneCommand) extends CommandMessage
+object SqsMessage extends ProtobufConversions {
+  final case class InvalidMessageType(className: String)
+    extends Exception(s"Invalid command message type $className")
+
+  def cast(msg: CommandMessage): Either[InvalidMessageType, SqsMessage] = msg match {
+    case ok: SqsMessage => Right(ok)
+    case invalid => Left(InvalidMessageType(invalid.getClass.getName))
+  }
+
+  def parseSqsMessage(message: Message): Either[Exception, SqsMessage] = {
+    for {
+      messageType <- SqsMessageType.fromMessage(message)
+      messageBytes <- Option(message.getBody)
+        .map(Right(_)).getOrElse(Left(EmptySqsMessageContents(message.getMessageId)))
+      contents <- Either.catchNonFatal(Base64.getDecoder.decode(messageBytes))
+      cmd <- Either.catchNonFatal {
+        messageType match {
+          case SqsRecordSetChangeMessage =>
+            fromPB(VinylDNSProto.RecordSetChange.parseFrom(contents))
+          case SqsZoneChangeMessage => fromPB(VinylDNSProto.ZoneChange.parseFrom(contents))
+        }
+      }
+    } yield SqsMessage(message.getReceiptHandle, cmd)
+  }
```

Comment:
handles all kinds of possible errors:

* The message type is not valid
* The message body is empty
* The message body was not valid Base 64
* The message body was valid Base 64, but protobuf serialization failed